org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/ImportManager.java

```diff
@@ -181,43 +182,81 @@ protected void appendType(final String qualifiedName, final String shortName, fi
 		} else if (needsQualifiedName(namespaceImport, shortName)) {
 			builder.append(qualifiedName);
 		} else {
-			if (imports.containsKey(outerShortName)) {
-				if (namespaceImport.equals(imports.get(outerShortName))) {
+			final Map<String, String> imps = internalGetImports();
+			if (imps.containsKey(outerShortName)) {
+				if (namespaceImport.equals(imps.get(outerShortName))) {
 					builder.append(shortName);
 				} else {
 					builder.append(qualifiedName);
 				}
 			} else {
-				imports.put(outerShortName, namespaceImport);
+				imps.put(outerShortName, namespaceImport);
 				builder.append(shortName);
 			}
 		}
 	}
 
 	protected boolean allowsSimpleName(String qualifiedName, String simpleName) {
-		return thisTypeQualifiedNames.contains(qualifiedName) || ((!thisCollidesWithJavaLang || !thisTypeSimpleNames.contains(simpleName)) && JAVA_LANG_PACK.matcher(qualifiedName).matches())
+		return getThisTypeQualifiedNames().contains(qualifiedName) || ((!thisCollidesWithJavaLang || !getThisTypeSimpleNames().contains(simpleName)) && JAVA_LANG_PACK.matcher(qualifiedName).matches())
 				|| equal(qualifiedName, simpleName);
 	}
 
 	protected boolean needsQualifiedName(String qualifiedName, String simpleName) {
-		return !organizeImports || (thisTypeSimpleNames.contains(simpleName) && !thisTypeQualifiedNames.contains(qualifiedName))
+		return !organizeImports || (getThisTypeSimpleNames().contains(simpleName) && !getThisTypeQualifiedNames().contains(qualifiedName))
 				|| CodeGenUtil2.isJavaLangType(simpleName);
 	}
 
 	public boolean addImportFor(JvmType type) {
-		final String qualifiedName = type.getQualifiedName(innerTypeSeparator);
+		final String qualifiedName = type.getQualifiedName(getInnerTypeSeparator());
 		final String simpleName = type.getSimpleName();
-		if (!allowsSimpleName(qualifiedName, simpleName) && !needsQualifiedName(qualifiedName, simpleName) && !imports.containsKey(simpleName)) {
-			imports.put(simpleName, qualifiedName);
+		final Map<String, String> imps = internalGetImports();
+		if (!allowsSimpleName(qualifiedName, simpleName) && !needsQualifiedName(qualifiedName, simpleName) && !imps.containsKey(simpleName)) {
+			imps.put(simpleName, qualifiedName);
 			return true;
 		}
 		return false;
 	}
 
 	public List<String> getImports() {
-		ArrayList<String> result = newArrayList(newLinkedHashSet(imports.values()));
+		ArrayList<String> result = newArrayList(newLinkedHashSet(internalGetImports().values()));
 		Collections.sort(result);
 		return result;
 	}
 	
+	/** Replies the internal set that contains the qualified names of "this" type.
```

Comment:
Stylistic remarks: please format javadoc with a linebreak after `/**`
since tag should state `@since 2.14`